staging/src/k8s.io/cluster-bootstrap/util/tokens/tokens.go

```diff
@@ -36,3 +40,64 @@ func ParseToken(s string) (tokenID, tokenSecret string, err error) {
 	}
 	return split[1], split[2], nil
 }
+
+// BootstrapTokenString is a token of the format abcdef.abcdef0123456789 that is used
+// for both validation of the practically of the API server from a joining node's point
+// of view and as an authentication method for the node in the bootstrap phase of
+// "kubeadm join". This token is and should be short-lived
+type BootstrapTokenString struct {
```

Comment:
If we keep this, can we move it to https://github.com/kubernetes/cluster-bootstrap/blob/master/token/api ?

Reply:
Current methods for `BootstrapTokenString` use several util functions. They are coupled. It is not a good pattern to import utils in api.

Reply:
In order to use `BootstrapTokenString` in a versioned API, it has to be versioned too. Therefore we need this, along with its interface implementation methods to be well versioned.
I can see the point, that we don't want to rely on much external stuff and I think, that we can safely avoid that while the types are versioned.

Reply:
This is not a serializable struct (it does not contain json tags for serialization), and it is not versioned (it doesn't have a "hub" internal type with specific serialization versions to allow for different representations in the future).